### PR TITLE
Update and modernise aqbanking

### PR DESCRIPTION
**Pull request by afh**
Note that aqbanking 5.5.1 should come along with the updated version of gwenhywfar (see #35471).

### CODE REVIEW

File: Library/Formula/aqbanking.rb [hunk 1]
```diff
@@ -31,17 +20,61 @@ def install
...
+      } # accountInfo
+    } # accountInfoList
     EOS
+    require "open3"
+    Open3.popen3("#{bin}/aqbanking-cli", "listbal") do |stdin, stdout, stderr|
+      stdin.write context
+      stdin.close
+      out = stdout.read
+      out =~ /^Account 110000000 000123456789  STRIPE TEST BANK            03.01.2014  13:00 1324.36 USD$/
+    end
```
Comment (mikemcquaid) [inline Library/Formula/aqbanking.rb]: Nice, thanks!

File: Library/Formula/aqbanking.rb [hunk 2]
```diff
@@ -31,17 +20,61 @@ def install
...
-    brew install aqbanking --devel
+      } # accountInfo
+    } # accountInfoList
     EOS
+    require "open3"
+    Open3.popen3("#{bin}/aqbanking-cli", "listbal") do |stdin, stdout, stderr|
+      stdin.write context
+      stdin.close
+      out = stdout.read
+      out =~ /^Account 110000000 000123456789  STRIPE TEST BANK            03.01.2014  13:00 1324.36 USD$/
```
Comment (jacknagel) [inline Library/Formula/aqbanking.rb]: Please use `assert_match pattern, stdout.read`; `=~` returns match-or-nil and [won't fail the test](https://github.com/Homebrew/homebrew/blob/f2324e12754b2040622d5696a2757a62c9e993c7/Library/Homebrew/cmd/test.rb#L59) properly.

Comment (afh) [inline Library/Formula/aqbanking.rb]: @jacknagel Thank you very much for this change proposal! Another useful tidbit I learned about Homebrew!

File: Library/Formula/aqbanking.rb [hunk 3]
```diff
@@ -31,17 +20,60 @@ def install
...
+        } #statusList
 
-    brew install aqbanking --devel --only-dependencies
-    brew install gwenhywfar --devel
-    brew install aqbanking --devel
+      } # accountInfo
+    } # accountInfoList
     EOS
+    require "open3"
+    Open3.popen3("#{bin}/aqbanking-cli", "listbal") do |stdin, stdout, _|
```
Comment (afh) [inline Library/Formula/aqbanking.rb]: I just saw that in the [libtiff](https://github.com/Homebrew/homebrew/blob/9f6926265f8e4be7cc80dfe9042f2cd3c1e8dc9e/Library/Formula/libtiff.rb) formula, that it is also possible to write testfiles.

Which approach is preferred by Homebrew?
The `popen3` one here of the `(testfiles/test).write` one similar to the on in libtiff?

Comment (mikemcquaid) [inline Library/Formula/aqbanking.rb]: Writing testfiles, thanks.

File: Library/Formula/aqbanking.rb [hunk 4]
```diff
@@ -31,17 +20,57 @@ def install
...
+          } #status
+        } #statusList
 
-    brew install aqbanking --devel --only-dependencies
-    brew install gwenhywfar --devel
-    brew install aqbanking --devel
+      } # accountInfo
+    } # accountInfoList
     EOS
+    assert_match /^Account\s+110000000\s+000123456789\s+STRIPE TEST BANK\s+03.01.2014\s+13:00\s+1324.36\s+USD\s+$/, shell_output("#{bin}/aqbanking-cli listbal -c #{context}")
```
Comment (tdsmith) [inline Library/Formula/aqbanking.rb]: Does this need to be in a particular timezone to match? You can set `ENV["TZ"]` in the test block.

Comment (afh) [inline Library/Formula/aqbanking.rb]: The things one needs to think about…
Many thanks @tdsmith!

**Comment by afh** [comment]
Thanks for merging

**Comment by mikemcquaid** [comment]
A pleasure, thanks for your work here.

**Pull request closed without merging**